src/test/java/net/wimpi/modbus/facade/ModbusTCPMasterTimeoutRegression/TimeoutRegressionTest.java

```diff
@@ -0,0 +1,39 @@
+package net.wimpi.modbus.facade.ModbusTCPMasterTimeoutRegression;
+
+import junit.framework.TestCase;
+import net.wimpi.modbus.ModbusCoupler;
+import net.wimpi.modbus.facade.ModbusTCPMaster;
+import net.wimpi.modbus.net.ModbusTCPListener;
+import net.wimpi.modbus.procimg.SimpleRegister;
+
+import java.net.InetAddress;
+import java.util.Arrays;
+
+public class TimeoutRegressionTest extends TestCase {
+  private static final int TEST_REGISTER_ADDRESS = 10000;
+  private static final int TEST_CLIENT_PORT = 12345;
+
+  public void testTimeout() throws Exception {
+    TestProcessImage spi = new TestProcessImage();
+    SlowRegister slowRegister = new SlowRegister(Arrays.asList(5000L));
+
+    spi.setRegister(TEST_REGISTER_ADDRESS, slowRegister);
+
+    ModbusCoupler.getReference().setProcessImage(spi);
+    ModbusCoupler.getReference().setMaster(false);
+    ModbusCoupler.getReference().setUnitID(0);
+
+    final ModbusTCPListener listener = new ModbusTCPListener(1, InetAddress.getLoopbackAddress());
+    listener.setPort(TEST_CLIENT_PORT);
+    listener.start();
+
+    final ModbusTCPMaster modbusMaster = new ModbusTCPMaster("127.0.0.1", 12345);
```

Comment:
```suggestion
    final ModbusTCPMaster modbusMaster = new ModbusTCPMaster("127.0.0.1", TEST_CLIENT_PORT);
```
Even better would be to use a dynamic port in order to avoid conflicts in CI environment.

Reply:
In theory, dynamic port would indeed be better, but in this case we can't get the port from the ModbusTCPListener as it does not properly synchronize it's member `m_ServerSocket`.

Reply:
Oh well. As we plan to invest as little as possible in this -- it is good enough.